rest/src/rest/RESTEmitter.cpp

```diff
@@ -262,12 +262,18 @@ auto zpt::RESTEmitter::has_pending(zpt::json _envelope) -> bool {
 auto zpt::RESTEmitter::reply(zpt::json _request, zpt::json _reply) -> void {
 	auto _exists = this->__pending.find(std::string(_request["channel"]));
 	if (_exists != this->__pending.end()) {
+		auto _callback = _exists->second;
+		this->__pending.erase(_exists);
 		if (_reply->ok()) {
-			auto _callback = _exists->second;
-			this->__pending.erase(_exists);
-			_callback(zpt::ev::Reply, std::string(_request["resource"]), _reply, this->self());
+			try {
+				_callback(zpt::ev::Reply, std::string(_request["resource"]), _reply, this->self());
+			} catch(zpt::assertion& _e) {
+				_callback(zpt::ev::Reply, std::string(_request["resource"]), { "status", _e.status(), "payload", { "text", _e.what(), "assertion_failed", _e.description(), "code", _e.code() } }, this->self());
+			} catch(std::exception& _e) {
+				_callback(zpt::ev::Reply, std::string(_request["resource"]), { "status", 500, "payload", { "text", _e.what(), "code", 0 } }, this->self());
+			}
 		} else {
```

Comment:
Good, agree that the callback should be notified if the channel eventually times out or gets disconnected.

Reply:
Tagged the wrong lines, was refering to the `else` branch!